### PR TITLE
Merge Charn's module to mainline

### PR DESCRIPTION
**Pull request by intaniger**

### CODE REVIEW

**Comment by intaniger** [comment]
I'll solve those conflicts and ask @chantmk to verify the result then.

File: .idea/terminal.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="TerminalProjectOptionsProvider">
+    <envs>
+      <env key="D:\Program Files\Android\Sdk\platform-t" value="ols=" />
```
Comment (chantmk) [inline .idea/terminal.xml]: What is this?... Seem dangerous lol

File: app/src/main/java/com/arfist/armona/screen/map/MapFragment.kt [hunk 2]
```diff
@@ -15,32 +17,37 @@ import androidx.fragment.app.activityViewModels
...
+    private var googleMap: GoogleMap? = null
     private val mapViewModel: MapViewModel by activityViewModels()
-
     private var isPermissionGranted = false
-
     private var direction: Direction? = null
-
     private var followLocation = false
 
+    // Temp
```
Comment (chantmk) [inline app/src/main/java/com/arfist/armona/screen/map/MapFragment.kt]: Removable

File: app/src/main/java/com/arfist/armona/screen/map/MapFragment.kt [hunk 3]
```diff
@@ -65,6 +72,9 @@ class MapFragment : Fragment() {
         return binding.root
     }
 
+    // Test
```
Comment (chantmk) [inline app/src/main/java/com/arfist/armona/screen/map/MapFragment.kt]: Removable

File: app/src/main/java/com/arfist/armona/screen/map/MapFragment.kt [hunk 4]
```diff
@@ -74,6 +84,28 @@ class MapFragment : Fragment() {
                 .navigate(MapFragmentDirections.actionMapFragmentToArFragment())
         }
 
+        binding.arViewModel = arViewModel
```
Comment (chantmk) [inline app/src/main/java/com/arfist/armona/screen/map/MapFragment.kt]: Removable

File: .idea/terminal.xml (hunk 1 above)
Comment (intaniger) [reply .idea/terminal.xml]: It's IDE's config and can be ignored. I'll delete it and add to .gitignore then.

**Review by chantmk — APPROVED** [review]
okie dokie

**Pull request merged**